packages/web-app-text-editor/src/index.js

```diff
@@ -19,41 +21,69 @@ const routes = [
   }
 ]
 
-const fileExtensionConfig = {
-  canBeDefault: true
+const fileExtensions = () => {
+  const extensions = [
+    {
+      extension: 'txt',
+      label: $gettext('Plain text file')
+    },
+    {
+      extension: 'md',
+      label: $gettext('Markdown file')
+    },
+    {
+      extension: 'js',
+      label: $gettext('JavaScript file')
+    },
+    {
+      extension: 'json',
+      label: $gettext('JSON file')
+    },
+    {
+      extension: 'xml',
+      label: $gettext('XML file')
+    },
+    {
+      extension: 'py',
+      label: $gettext('Python file')
+    },
+    {
+      extension: 'php',
+      label: $gettext('PHP file')
+    },
+    {
+      extension: 'yaml',
+      label: $gettext('YAML file')
+    }
+  ]
+  let blackList =
+    window.Vue.$store.getters.extensionConfigByAppId(appId).blockedFileExtensions || []
+  if (typeof blackList === 'string') {
+    blackList = [blackList]
+  }
+  return extensions.reduce((acc, extensionItem) => {
+    if (!blackList.includes(extensionItem.extension)) {
+      acc.push(extensionItem)
+    }
+    return acc
+  }, [])
 }
 
 const appInfo = {
   name: $gettext('Text Editor'),
-  id: 'text-editor',
+  id: appId,
   icon: 'file-text',
   isFileEditor: true,
-  extensions: [
-    {
-      extension: 'txt',
+  extensions: fileExtensions().map((extensionItem) => {
+    return {
+      extension: extensionItem.extension,
       newFileMenu: {
         menuTitle($gettext) {
```

Comment:
Since we have all file extensions in the same level (in the "new" menu), the fact that we allow creating this many types will generate a huge list. I prefer the previous behavior: we can open many, but create just txt  - users can still rename the extension. Ofc it would be even better if we could just open any text mimetype instead of file extensions.

Reply:
Updated, could you please check out if the new behaviour is better for your needs?

Reply:
I guess you still need to set `canBeDefault` to false explicitly? Other wise, [this](https://github.com/owncloud/web/blob/master/packages/web-runtime/src/store/apps.js#L57).

Reply:
👍